crates/polars-plan/src/plans/optimizer/predicate_pushdown/mod.rs

```diff
@@ -378,7 +378,7 @@ impl<'a> PredicatePushDown<'a> {
                                 }
                                 scan_type.remove_metadata();
                             }
-                            if paths.is_empty() {
+                            if new_paths.is_empty() {
```

Comment:
I discovered there was other code in place to ensure the invariant that a reader always has at least one path here, it would replace the scan node with a `DataFrameScan` but it wasn't being hit because it was incorrectly checking the old `paths` instead of `new_paths`. This changes the fix for https://github.com/pola-rs/polars/pull/12575